### PR TITLE
Feature/auth update

### PR DESCRIPTION
**Pull request by MaxAstin**

### CODE REVIEW

File: app/src/main/java/com/bunbeauty/papakarlo/feature/auth/ConfirmFragment.kt [hunk 1]
```diff
@@ -134,7 +149,7 @@ class ConfirmFragment : BaseFragment(R.layout.layout_compose) {
             Text(
                 modifier = Modifier
                     .padding(top = FoodDeliveryTheme.dimensions.smallSpace),
-                text = stringResource(R.string.msg_confirm_phone_info) + confirmState.phoneNumber,
+                text = stringResource(R.string.msg_confirm_phone_info) + state.phoneNumber,
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/papakarlo/feature/auth/ConfirmFragment.kt:152]: Так мы вроде хотели в параметры stringResource засовывать через %s?

File: app/src/main/java/com/bunbeauty/papakarlo/feature/auth/LoginFragment.kt [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .verticalScroll(rememberScrollState())
+                    .padding(FoodDeliveryTheme.dimensions.mediumSpace),
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.Center
+            ) {
+                BoxWithConstraints {
+                    if (maxHeight > 200.dp) {
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/papakarlo/feature/auth/LoginFragment.kt:85]: 200 lets move to cons like MAX_LOGO_SIZE?

File: shared/src/androidMain/kotlin/com/bunbeauty/shared/data/FirebaseAuthRepository.kt [hunk 3]
```diff
@@ -1,17 +1,9 @@
 package com.bunbeauty.shared.data
 
-import com.google.firebase.auth.FirebaseAuth
+actual class FirebaseAuthRepository {
```
Comment (xidealo) [inline shared/src/androidMain/kotlin/com/bunbeauty/shared/data/FirebaseAuthRepository.kt:3]: Lets remove if we don't need it

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/confirm/ConfirmViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    private var direction: SuccessLoginDirection? = null
+
+    init {
+        startResendTimer()
+    }
+
+    override fun handleAction(action: Confirm.Action) {
+        when (action) {
+            is Confirm.Action.Init -> {
+                init(action.direction, action.phoneNumber)
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/confirm/ConfirmViewModel.kt:43]: add name of params

File: shared/src/androidMain/kotlin/com/bunbeauty/shared/data/FirebaseAuthRepository.kt (hunk 3 above)
Comment (MaxAstin) [reply shared/src/androidMain/kotlin/com/bunbeauty/shared/data/FirebaseAuthRepository.kt:3]: We need it until update in iOS, it's `actual class`

File: app/src/main/java/com/bunbeauty/papakarlo/feature/auth/LoginFragment.kt (hunk 2 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/auth/LoginFragment.kt:85]: For Compose conts we should use `MaxLogoSize`

File: app/src/main/java/com/bunbeauty/papakarlo/feature/auth/ConfirmFragment.kt (hunk 1 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/auth/ConfirmFragment.kt:152]: fixed

File: app/src/main/java/com/bunbeauty/papakarlo/feature/auth/LoginFragment.kt (hunk 2 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/auth/LoginFragment.kt:85]: fixed

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/confirm/ConfirmViewModel.kt (hunk 4 above)
Comment (MaxAstin) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/confirm/ConfirmViewModel.kt:43]: fixed

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/login/FormatPhoneNumberUseCase.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package com.bunbeauty.shared.domain.feature.login
+
+import com.bunbeauty.shared.Constants
+
+class FormatPhoneNumberUseCase {
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/login/FormatPhoneNumberUseCase.kt:5]: need tests

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/login/GetPhoneNumberCursorPositionUseCase.kt [hunk 6]
```diff
@@ -0,0 +1,51 @@
+package com.bunbeauty.shared.domain.feature.login
+
+class GetPhoneNumberCursorPositionUseCase(
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/login/GetPhoneNumberCursorPositionUseCase.kt:3]: need tests

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/use_case/auth/RequestCodeUseCase.kt [hunk 7]
```diff
@@ -0,0 +1,28 @@
+package com.bunbeauty.shared.domain.use_case.auth
+
+import com.bunbeauty.shared.domain.exeptions.InvalidPhoneNumberException
+import com.bunbeauty.shared.domain.exeptions.SomethingWentWrongException
+import com.bunbeauty.shared.domain.exeptions.TooManyRequestsException
+import com.bunbeauty.shared.domain.model.CodeResponse
+import com.bunbeauty.shared.domain.repo.AuthRepo
+
+class RequestCodeUseCase(
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/use_case/auth/RequestCodeUseCase.kt:9]: need tests

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/login/LoginViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+class LoginViewModel(
+    private val requestCode: RequestCodeUseCase,
+    private val formatPhoneNumber: FormatPhoneNumberUseCase,
+    private val getPhoneNumberCursorPosition: GetPhoneNumberCursorPositionUseCase,
+) : SharedStateViewModel<Login.State, Login.Action, Login.Event>(Login.State()) {
+
+    override fun handleAction(action: Login.Action) {
+        when (action) {
+            Login.Action.Init -> init()
+            is Login.Action.ChangePhoneNumber -> updatePhoneNumber(action.phoneNumber, action.cursorPosition)
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/login/LoginViewModel.kt:20]: name of params

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/login/LoginViewModel.kt [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        state { state ->
+            state.copy(isLoading = false)
+        }
+    }
+
+    private fun updatePhoneNumber(phoneNumber: String, cursorPosition: Int) {
+        state { state ->
+            state.copy(
+                phoneNumber = formatPhoneNumber(phoneNumber),
+                phoneNumberCursorPosition = getPhoneNumberCursorPosition(phoneNumber, cursorPosition)
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/login/LoginViewModel.kt:37]: name of params

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/login/FormatPhoneNumberUseCase.kt (hunk 5 above)
Comment (MaxAstin) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/login/FormatPhoneNumberUseCase.kt]: ✅

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/login/GetPhoneNumberCursorPositionUseCase.kt (hunk 6 above)
Comment (MaxAstin) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/login/GetPhoneNumberCursorPositionUseCase.kt]: ✅

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/use_case/auth/RequestCodeUseCase.kt (hunk 7 above)
Comment (MaxAstin) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/use_case/auth/RequestCodeUseCase.kt]: ✅

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/login/LoginViewModel.kt (hunk 8 above)
Comment (MaxAstin) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/login/LoginViewModel.kt]: ✅

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/login/LoginViewModel.kt (hunk 9 above)
Comment (MaxAstin) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/login/LoginViewModel.kt]: ✅

**Pull request merged**